src/main/java/com/github/manolo8/darkbot/core/manager/StatsManager.java

```diff
@@ -243,4 +256,63 @@ public double getTotalHonor() {
     public double getEarnedHonor() {
         return earnedHonor;
     }
+
+    public AverageStats getCpuStats() {
+        return cpuStat;
+    }
+
+    public AverageStats getPingStats() {
+        return pingStat;
+    }
+
+    public AverageStats getTickStats() {
+        return tickStat;
+    }
+
+    public AverageStats getMemoryStats() {
+        return memoryStat;
+    }
+
+    public static class AverageStats {
+        private static final DecimalFormat MAX_TWO_PLACES_FORMAT = new DecimalFormat("0.##");
+
+        private int count;
+        private double last, average, min = Double.MAX_VALUE, max = Double.MIN_VALUE;
+
+        private Consumer<String> onChange;
+
+        public void accept(double value) {
+            max = Math.max(max, value);
+            min = Math.min(min, value);
+
+            if (last != value && onChange != null)
+                onChange.accept(MAX_TWO_PLACES_FORMAT.format(value));
+            last = value;
+
+            if (count == Integer.MAX_VALUE) {
+                count = 0;
+                average = 0;
+            }
+
+            average = average + (value - average) / (++count);
+        }
```

Comment:
I'm not convinced that keeping "permanent" stats is useful. The min/average/max over a long period doesn't seem too relevant, and i think for display purposes it makes more sense to user "rolling averages" or "weighted averages" where the most recent data is the one that matters the most.

That's essentially what tick-time used to do, where it would take the new sample as 1/10th while all past samples were 9/10th. This means that what the data will reflect most is the past few ticks/updates, as they "each" will have about 10%, while older data will become irrelevant over time.

I believe this class needs similar behavior, for all 3 min, avg and max.